### PR TITLE
Source Package Generation in CI Pipeline

### PR DESCRIPTION
**Pull request by nickclark2016**
**What does this PR do?**

Adds a job to generate source packages.  This job depends on the Linux, Windows, and MacOSX jobs completing successfully, as these should be a prerequisite to building source packages (why build useless source packages?).

This also helps to automate a step of the Premake release process, where we'd be doing this manually.

**How does this PR change Premake's behavior?**

N/A

**Anything else we should know?**

N/A

**Did you check all the boxes?**

- [x] Focus on a single fix or feature; remove any unrelated formatting or code changes
- [x] Add unit tests showing fix or feature works; all tests pass
- [x] Mention any [related issues](https://github.com/premake/premake-core/issues) (put `closes #XXXX` in comment to auto-close issue when PR is merged)
- [x] Follow our [coding conventions](https://github.com/premake/premake-core/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] Minimize the number of commits
- [x] Align [documentation](https://github.com/premake/premake-core/tree/master/website) to your changes

*You can now [support Premake on our OpenCollective](https://opencollective.com/premake). Your contributions help us spend more time responding to requests like these!*

### CODE REVIEW

File: .github/workflows/ci-workflow.yml [hunk 1]
```diff
@@ -70,3 +73,26 @@ jobs:
...
+    - name: Checkout # Need to checkout to get the package action
+      uses: actions/checkout@v2
+    - name: Download Release
+      uses: actions/download-artifact@v3
+      with:
+        name: premake-linux-${{ matrix.platform }}
+    - name: Switch Branches
+      run: git switch -C ${{ github.ref_name }}
+    - name: Generate Sources
+      run: chmod 555 ./premake5 && ./premake5 package $BRANCH source force
```
Comment (KyrietS) [inline .github/workflows/ci-workflow.yml:93]: Should it be `$BRANCH_NAME` here? 🤠

File: .github/workflows/ci-workflow.yml [hunk 2]
```diff
@@ -70,3 +73,26 @@ jobs:
...
+    needs: [linux, macosx, windows] # Pass all builds before creating source bundle
+    steps:
+    - name: Checkout # Need to checkout to get the package action
+      uses: actions/checkout@v2
+    - name: Download Release
+      uses: actions/download-artifact@v3
+      with:
+        name: premake-linux-${{ matrix.platform }}
+    - name: Switch Branches
+      run: git switch -C ${{ github.ref_name }}
```
Comment (KyrietS) [inline .github/workflows/ci-workflow.yml:91]: Can you explain the purpose of this switching?

Comment (nickclark2016) [reply .github/workflows/ci-workflow.yml:91]: We're trying to figure out how to get this to pull down the right branch in the package script.  It was an idea that @samsinsane and I had discussed offline, but I'm honestly blanking on how to get this working.  The `git clone` command in the package script isn't playing nice.  If you've got ideas, I'd love to hear them, I'm blanking.

File: scripts/package.lua [hunk 3]
```diff
@@ -47,7 +47,7 @@
 		error("Unsupported host os", 0)
 	end
 
-	local usage = 'usage is: package <branch> <type> [<compiler>]\n' ..
+	local usage = 'usage is: package <branch> <type> [<compiler>] [force]\n' ..
 		'       <branch> is the name of the release branch to target\n' ..
 		'       <type> is one of "source" or "binary"\n' ..
 		'       <compiler> (default: ' .. allowedCompilers[1] .. ') is one of ' .. table.implode(allowedCompilers, "", "", " ")
```
Comment (Jarod42) [inline scripts/package.lua:53]: Missing doc/help for `force` flag.

Comment (nickclark2016) [reply scripts/package.lua:53]: Good catch. I'll add that in my next commit. Still wrapping my head around how to get this thing working for PRs across repos.  It worked in my repo, but didn't work after I made the PR.

Comment (nickclark2016) [reply scripts/package.lua:53]: Follow up. We have no documentation for this file at all. We can add it though if you think we need it.

Comment (Jarod42) [reply scripts/package.lua:53]: I meant `usage` string "explains" quickly `<branch>` `<type>` `<compiler>` but not `[force]` (as non interactive).
Not really important...

Comment (samsinsane) [reply scripts/package.lua:53]: Looks like the latest commit doesn't have this change @nickclark2016?

**Pull request closed without merging**